src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.html

```diff
@@ -68,23 +68,55 @@ <h4 jhiTranslate="artemisApp.fileUploadAssessment.dashboard.heading" [translateV
                     <td>
                         <span *jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_TA']">
                             <button
-                                *ngIf="submission && !submission.result"
+                                *ngIf="submission && !submission.result && exercise.course"
                                 [routerLink]="['/course-management', exercise.course?.id, 'file-upload-exercises', exercise.id, 'submissions', submission.id, 'assessment']"
                                 [disabled]="busy"
                                 class="btn btn-outline-secondary btn-sm mb-1"
                             >
                                 <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.assess' | translate }}
                             </button>
+                            <button
+                                *ngIf="submission && !submission.result && exercise.exerciseGroup"
+                                [routerLink]="[
+                                    '/course-management',
+                                    exercise.exerciseGroup?.exam.course.id,
+                                    'file-upload-exercises',
+                                    exercise.id,
+                                    'submissions',
+                                    submission.id,
+                                    'assessment'
+                                ]"
+                                [disabled]="busy"
+                                class="btn btn-outline-secondary btn-sm mb-1"
+                            >
+                                <fa-icon [icon]="'folder-open'" [fixedWidth]="true"></fa-icon>&nbsp;{{ 'artemisApp.assessment.dashboard.actions.assess' | translate }}
+                            </button>
                         </span>
                         <span *jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_TA']">
                             <button
-                                *ngIf="submission && submission.result"
+                                *ngIf="submission && submission.result && exercise.course"
```

Comment:
Just a minor suggestion: To improve the readability of the code and reduce the checks done you can wrap these in ngIf*"abc else xyz" blocks. The else block can then be contained within a <ng-template #xyz>
@TobiasPr just showed me this a few days ago and I thought id share :)